uat/custom-components/client-java-sdk/src/main/java/com/aws/greengrass/testing/mqtt5/client/sdkmqtt/MqttConnectionImpl.java

```diff
@@ -586,7 +589,7 @@ private List<UserProperty> convertToUserProperties(@NonNull List<Mqtt5Properties
         List<UserProperty> userProperties = new ArrayList<>();
         properties.forEach(p -> {
             userProperties.add(new UserProperty(p.getKey(), p.getValue()));
-            logger.atInfo().log("{} MQTT userProperties: {}, {}", type, p.getKey(), p.getValue());
+            logger.atInfo().log("{} Tx user property '{}':'{}'", type, p.getKey(), p.getValue());
```

Comment:
Line 308
publishPacket.withUserProperties(convertToUserProperties(userProperties, "TX PUBLISH"));

already contains "TX".
I suggest delete it from line 308.
Also check RX publish.

And the same situation for DISCONNECT

Reply:
Yes, looks Tx will doubled. Because that method is called only to convert from MQTT to gRPC that means can used only on RX side. I will remove rx or tx from string literals passed to both convertToUserProperties() and convertToMqtt5Properties() and add Tx and Rx directly to methods to avoid misusage.

Because Michael already merged that PR I going join with next PR